internal/driver/glfw/loop.go

```diff
@@ -65,24 +65,40 @@ func runOnMainWithWait(f func(), wait bool) {
 var refreshingCanvases []fyne.Canvas
 
 func (d *gLDriver) drawSingleFrame() {
+	refreshed := false
 	for _, win := range d.windowList() {
 		w := win.(*window)
+		if w.closing {
+			continue
+		}
+
 		canvas := w.canvas
-		closing := w.closing
-		visible := w.visible
 
 		// CheckDirtyAndClear must be checked after visibility,
 		// because when a window becomes visible, it could be
 		// showing old content without a dirty flag set to true.
 		// Do the clear if and only if the window is visible.
-		if closing || !visible || !canvas.CheckDirtyAndClear() {
+		if !w.visible || !canvas.CheckDirtyAndClear() {
+			// Window hidden or not being redrawn, mark canvasForObject
+			// cache alive if it hasn't been done recently
+			// n.b. we need to make sure threshold is a bit *after*
+			// time.Now() - CacheDuration()
+			threshold := time.Now().Add(-cache.CacheDuration()).Add(1 * time.Second)
```

Comment:
I think it is a bit clearer to do `.Sub().Add()` or even the other way around?

Reply:
Hmm. That's strange. Do we get strange behaviour if we do `.Add(time.Second - cache.CacheDuration())`?

Reply:
That works fine, switched to it